### PR TITLE
feat: adds CodeQL

### LINKED ISSUE #215: add codeql
**Describe the request**

**Explain why KMS Plugin for Key Vault needs it**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by nilekhc**
**Reason for Change**:

**Issue Fixed**:

fixes #215 

**Notes for Reviewers**:

### CODE REVIEW

**Comment by nilekhc** [comment]
cc @sozercan

File: .github/workflows/codeql.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@04df1262e6247151b5ac09cd2c303ac36ad3f62b
```
Comment (aramase) [inline .github/workflows/codeql.yaml:27]: Latest is `b2c19fb9a2a485599ccf4ed5d65527d94bc57226` for tag `v2.3.0`

**Review by aramase — APPROVED** [review]

**Review by aramase — APPROVED** [review]

**Pull request merged**